src/models.ts

```diff
@@ -297,3 +304,33 @@ export class AdvancedFilter extends Filter {
     return filter;
   }
 }
+
+export interface IDataReference {
+  target: IFilterTarget;
+}
+
+export interface IEqualsDataReference extends IDataReference {
+  equals: string | boolean | number;
+}
+
+export interface IBetweenDataReference extends IDataReference {
+  between: (string | boolean | number)[];
+}
+
+export interface IValueDataReference extends IDataReference {
+  value: string | boolean | number;
```

Comment:
If it is only optional because of our development timelines not because it's actually optional right?  I don't think this should influence how we write the interface.  It seems safer to remove and add later.

If it's optional forever then users would still have to check for it's existence before trying to operate on it which isn't nice either.